### PR TITLE
feat: Sync users with algolia

### PR DESCRIPTION
**Pull request by peterstarling**
https://trello.com/c/57EivHKQ/1795-prepare-sync-of-users-between-squidex-and-algolia

### CODE REVIEW

File: apps/asap-server/scripts/cli.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+          choices: ['users', 'research-outputs'],
+          demandOption: true,
+        })
+        .option('filename', {
+          alias: 'f',
+          type: 'string',
+          description: 'The output file name',
+        }),
+    handler: async ({ entity, filename }) =>
+      exportEntity(entity as 'users' | 'research-outputs', filename),
```
Comment (niezgoda) [inline apps/asap-server/scripts/cli.ts:25]: Could you move this type somewhere? You are using it again in export-entity.ts

File: apps/asap-server/scripts/export-entity.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  const file = await fs.open(filename || `${entity}.json`, 'w');
+
+  let recordCount = 0;
+  let total: number;
+  let records: ListResponse<unknown>;
+  let page = 1;
+
+  await file.write('[\n');
+
+  do {
```
Comment (niezgoda) [inline apps/asap-server/scripts/export-entity.ts:25]: this loop could be in some helper :)

**Review by niezgoda — CHANGES_REQUESTED** [review]

File: apps/asap-server/scripts/cli.ts (hunk 1 above)
Comment (kmaid) [reply apps/asap-server/scripts/cli.ts:25]: Could you potentially type it in line 8?

**Comment by peterstarling** [comment]
i love it how nobody noticed that this is introducing two different indices where we clearly said it would be just one

**Comment by peterstarling** [comment]
apologies -  @kmaid noticed

**Comment by kmaid** [comment]
image

**Pull request closed without merging**